types/mapcmd/map_it_test.go

```diff
@@ -0,0 +1,448 @@
+package mapcmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+	"github.com/hazelcast/hazelcast-go-client/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/hazelcast/hazelcast-commandline-client/internal/it"
+)
+
+// todo add value-file test
+func TestMapPut(t *testing.T) {
+	it.MapTesterWithConfigAndMapName(t, func(t *testing.T, c *hazelcast.Config, m *hazelcast.Map, mn string) {
+		mapNameArg := fmt.Sprintf(` --name %s`, mn)
```

Comment:
`mapNameArg` is used quite a bit in other tests too. I would add the `WithName` method to `it.MapTestContext` and use it instead of `it.MapTester`. Maybe something like:
```
func (tcx MapTestContext) WithName(args string) {
     return fmt.Sprintf(`--name %s %s`, tcx.MapName, args)
}
```
Then:
```
func TestMapPut(t *testing.T) {
    tcx := it.MapTestContext{T: t}
    tcx.Tester(func(tcx it.MapTestContext) {
                t := tcx.T
                m := tcx.M
		tcs := []struct {
			name        string
			args        string
			key         interface{}
			value       interface{}
			errContains string
		}{
			{
				name:  "valid put key(string), value(string)",
				args:  tcx.WithName("--key k1 --value v1"),
				key:   "k1",
				value: "v1",
			},
   // ...    
```

Reply:
Good idea, did something a bit different https://github.com/hazelcast/hazelcast-commandline-client/pull/114/commits/028b0d911ab55d5b095c7ab97bd6d8cf0da85bbc